src/mscorlib/src/System/Globalization/CultureInfo.cs

```diff
@@ -920,7 +893,9 @@ public virtual DateTimeFormatInfo DateTimeFormat
 
         public void ClearCachedData()
         {
-            Init(); // reset the default culture values
+            // reset the default culture values
+            s_userDefaultCulture = GetUserDefaultCulture();
```

Comment:
Wouldn't it be better to just reset these to null?
That way, these will again be lazily initialized if **UserDefaultUICulture** or **InstalledUICulture** are used again.
If however they are not used again (before the next invocation of `ClearCachedData()`), you're saving some cycles.

Note that you'd have to modify `InitializeUserDefaultCulture()` and `InitializeUserDefaultUICulture()` to take this into account.

Reply:
ClearCachedData method is very rarely used. I think that the current simple and less efficient implementation is the right trade-off.

Thanks for the feedback anyway.